doc/md/fundamentals/2-types/12-advanced-types.md

```diff
@@ -58,14 +57,16 @@ public type List<T> = ?(T, List<T>);
 
 Reversing a linked list involves iterating through the list and prepending each element to a new list. This approach demonstrates list traversal and structural mutation without using the built-in `reverse` method.
```

Comment:
```suggestion
Reversing a linked list involves iterating through the list and prepending each element to a new list. This approach demonstrates list traversal without using `List.reverse` library function.
```
The list itself isn't being mutated and can't be. You just some local state to store the new list under construction.